### PR TITLE
Copy files from todo list app

### PR DESCRIPTION
**Pull request by Ibnballo1**
I did the following in this pull request;

- I created a new empty repo.
- I copied my code from the "To Do list" project.
- I add and commit it as one big chunk of code in a feature branch.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import './style.css';
+import MyTasks from './modules/forTask.js';
+
+const formAddItems = document.getElementById('formAddItems');
+const addDescription = document.getElementById('toAdd');
+const listTitle = document.getElementById('listTitle');
+const clearTask = document.getElementById('clearTask');
+
+let dataCount = 0;
```
Comment (joffreynk) [inline src/index.js:9]: kindly fix the line to an array to avoid type conversion

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+const getStorage = localStorage.getItem('dataCount');
```
Comment (joffreynk) [inline src/index.js:10]: you are having multiple data in local storage, kindly change it to an array and put all data as object

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    eachList.selectTask(newTask);
+  });
+  description.addEventListener('keydown', () => {
+    eachList.selectTask(newTask);
+  });
+  checkbox.addEventListener('change', () => {
+    eachList.updateCheckbox(newTask);
+    eachList.checkTask(newTask.id);
+  });
+}
```
Comment (joffreynk) [inline src/index.js]: avoid using for loop try forEach method

Comment (joffreynk) [inline src/index.js]: duplicate events

File: src/modules/forTask.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    const description = document.getElementById(`input${id}`);
+    description.toggleAttribute('disabled');
+    description.classList.toggle('done');
+  };
+
+  // local storage
+  setData = () => {
+    localStorage.setItem('dataTasks', JSON.stringify(this.addedItems));
+    localStorage.setItem('dataCount', this.taskIndex);
+  };
```
Comment (joffreynk) [inline src/modules/forTask.js:122]: use array of objects to wrap all the data in

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  mode: 'development',
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist'),
+    clean: true,
+  },
```
Comment (joffreynk) [inline webpack.config.js]: please, entry and output are not necessary

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    rules: [
+      {
+        test: /\.css$/i,
+        use: ['style-loader', 'css-loader'],
+      },
+    ],
+  },
+  devServer: {
+    static: './dist',
+  },
```
Comment (joffreynk) [inline webpack.config.js]: try to remove the lines also webpack will handle them for you

File: src/style.css [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+.each-task button {
+  font-size: 18px;
+}
+
+.completed {
+  background-color: #f6f6f6;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
```
Comment (joffreynk) [inline src/style.css:132]: completed activity should have line-through property to show completed activity

**Review by joffreynk — CHANGES_REQUESTED** [review]
## Good Job so far but still some changes to work on
your design looks good but fix these changes

File: src/index.js (hunk 1 above)
Comment (Ibnballo1) [reply src/index.js:9]: I don't quite understand what you mean here. Please elaborate.

File: src/index.js (hunk 2 above)
Comment (Ibnballo1) [reply src/index.js:10]: Please what are the multiple data in the localStorage as I can't see multiple data from here?

File: src/modules/forTask.js (hunk 4 above)
Comment (joffreynk) [reply src/modules/forTask.js:122]: sorry that comment supposed to be here

File: src/index.js (hunk 2 above)
Comment (joffreynk) [reply src/index.js:10]: here you change datacount from integer to an array

File: src/index.js (hunk 3 above)
Comment (Ibnballo1) [reply src/index.js]: I am yet to figure out how I can use `forEach()` method for this, as it seems forEach can only be used on an array.

Comment (Ibnballo1) [reply src/index.js]: I don't think this should be considered a duplicate as they are all having different events.

File: src/modules/forTask.js (hunk 4 above)
Comment (Ibnballo1) [reply src/modules/forTask.js:122]: how do you think this can be achieved?

File: src/index.js (hunk 3 above)
Comment (joffreynk) [reply src/index.js]: here you are trying to display data from local storage and we are saving data in form of arrays
apply forEach to addedItems

File: webpack.config.js (hunk 5 above)
Comment (Ibnballo1) [reply webpack.config.js]: Why is it not necessary as in the requirement, Microverse never said it isn't necessary?

File: src/style.css (hunk 7 above)
Comment (Ibnballo1) [reply src/style.css:132]: Thanks, will work on this.

File: webpack.config.js (hunk 5 above)
Comment (joffreynk) [reply webpack.config.js]: you don't need configuration if you don't change the folder path. it is the one which causes your deployment to deploy some codes and ignore other codes

File: src/modules/forTask.js (hunk 4 above)
Comment (joffreynk) [reply src/modules/forTask.js:122]: you need to set an array that works on the whole program the form of your array is like this: 
`let data = [
  {
    index: 5,
    completed: false,
    description: 'do some sports',
  },
  {
    index: 2,
    completed: false,
    description: 'check messages',
  },
  {
    index: 4,
    completed: false,
    description: 'clean house ',
  },
  {
    index: 1,
    completed: false,
    description: 'wake up at 6am',
  },
  {
    index: 6,
    completed: false,
    description: 'take shower',
  },
  {
    index: 3,
    completed: false,
    description: 'wash dishes',
  },
  {
    index: 7,
    completed: false,
    description: 'start daily activities',
  },
];`

File: src/style.css (hunk 7 above)
Comment (Ibnballo1) [reply src/style.css:132]: And this part of the code is not for the checked box.

File: src/index.js (hunk 3 above)
Comment (joffreynk) [reply src/index.js]: check down in the code you have the same events
https://github.com/Ibnballo1/ToDo-List-Review/blob/54837c672f25906d8a7377bbb7887c1f2ad3428e/src/index.js#L36-L51

and here 

https://github.com/Ibnballo1/ToDo-List-Review/blob/54837c672f25906d8a7377bbb7887c1f2ad3428e/src/index.js#L62-L76

File: src/style.css (hunk 7 above)
Comment (joffreynk) [reply src/style.css:132]: now you put it on complete part because it is not there

File: src/index.js (hunk 3 above)
Comment (Ibnballo1) [reply src/index.js]: not working, I am yet to figure it out

File: src/modules/forTask.js (hunk 4 above)
Comment (Ibnballo1) [reply src/modules/forTask.js:122]: This will be a lot of work.

**Review by joffreynk — APPROVED** [review]
your project is complete now you can merge it

**Pull request merged**